Added form update guide

ankita240796 opened the pull request:
closes #369 

## What is included in this PR?
Added form update guide which describes:
- Cases where version update is needed
- Cases where a new form is needed

lognaturel commented:
Fantastic, thanks for taking this one on, @ankita240796! One thing to add to the list of allowed changes is changing the order of questions. It's also possible to add translations. I initially thought that was implied by changing translations but it might be better to make it explicit. It's also possible to update media like CSVs or images.

It would also be good to mention the behavior on the Collect side. Collect will treat different versions of the same form completely independently. It won't notify the users of the existence of new versions. That behavior is about to change thanks to some features @grzesiek2010 is working on but we should start with documenting the current behavior.

@grzesiek2010 anything else you'd like to see added?

Including full form examples is a new thing. I'm not sure how I feel about it! On one hand, it is nice to be able to try things out right away and your examples are cool! On the other it's one more thing to maintain and manage. Targeted, inline examples usually feel like enough to me.

It would be good for you and @adammichaelwood to discuss options for where this document should go.

ankita240796 commented:
Thank you @lognaturel!
I added the information about the order of questions, translations, updating media and Collect behavior with updated forms.

I feel full form examples are good for clear understanding. I added them to form-interaction as well. Since they are not really large examples, we can maintain them with little or no issue. What do you think?

grzesiek2010 commented on form-update.rst:
```diff
@@ -0,0 +1,140 @@
+   :alt: Image showing new version 2017120708 selected in the list of forms to be downloaded.
+   :class: device-screen-vertical
+
+  Both versions of the form exist in the device of the user and the user will be allowed to fill a older version and submit the form to the Aggregate server. 
```
an older*

grzesiek2010 replied:
an older*

grzesiek2010 replied:
an older*

grzesiek2010 commented on form-update.rst:
```diff
@@ -0,0 +1,140 @@
+
+- Changing the text or translation of a label
+- Changing validations, calculations, relevants
+- Changing options for a select_one or select_multiple
```
Is that about external csv files too or only about options that we store in a form? Because we can edit csv files too like media files.

ankita240796 replied:
I checked this and we can update the options in external csv files too but in that case, we don't need to update the version number.

grzesiek2010 commented:
@ankita240796 @lognaturel maybe it would be good o mention that we don't need to use (or update) version number if we need to change only media files or csv files. I mean if your changes are not in a form you don't need to change the version number.

ankita240796 commented:
Thanks for the review @grzesiek2010! I made the requested changes.

lognaturel commented on form-update.rst:
```diff
@@ -0,0 +1,145 @@
+- Changing options for a select_one or select_multiple
+- Changing the order of questions
+- Adding translations
+- Update media for your form
```
To match the rest in the list this should start with "Updating" rather than "Update". I think you could make this something like "Updating media including CSVs"

lognaturel commented on form-update.rst:
```diff
@@ -0,0 +1,145 @@
+
+.. note::
+
+  - If you update media for your form, you don't need to change the version.
```
"only update"?

ankita240796 replied:
I made this statement clear in the recent commit.

adammichaelwood commented:
> adding full form examples 

#386 will cover that when we get to it, and this is informative now, so I think it's good to merge

> and on where the page should go

The form building TOC is good for now.
